src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -62,36 +62,35 @@ GcsPlacementGroupManager::GcsPlacementGroupManager(
     std::shared_ptr<GcsPlacementGroupSchedulerInterface> scheduler,
     std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage)
     : gcs_placement_group_scheduler_(std::move(scheduler)),
-      gcs_table_storage_(gcs_table_storage),
+      gcs_table_storage_(std::move(gcs_table_storage)),
       reschedule_timer_(io_context) {}
 
 void GcsPlacementGroupManager::RegisterPlacementGroup(
-    const ray::rpc::CreatePlacementGroupRequest &request,
-    std::function<void(std::shared_ptr<GcsPlacementGroup>)> callback) {
+    const ray::rpc::CreatePlacementGroupRequest &request, EmptyCallback callback) {
   RAY_CHECK(callback);
   const auto &placement_group_spec = request.placement_group_spec();
   auto placement_group_id =
       PlacementGroupID::FromBinary(placement_group_spec.placement_group_id());
-
   auto placement_group = std::make_shared<GcsPlacementGroup>(request);
+
+  // TODO(ffbin): If GCS is restarted, GCS client will repeatedly send
+  // `CreatePlacementGroup` requests,
+  //  which will lead to resource leakage, we will solve it in next pr.
   // Mark the callback as pending and invoke it after the placement_group has been
   // successfully created.
-  placement_group_to_register_callbacks_[placement_group_id].emplace_back(
-      std::move(callback));
-  RAY_CHECK(registered_placement_groups_
-                .emplace(placement_group->GetPlacementGroupID(), placement_group)
-                .second);
+  placement_group_to_register_callbacks_[placement_group_id] = std::move(callback);
```

Comment:
In this case, can we rename it to `placement_group_to_register_callback_ `?

Reply:
fixed